src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTabRelatedArticles.java

```diff
@@ -0,0 +1,169 @@
+/**
+ *
+ */
+package net.sf.jabref.gui.entryeditor;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javax.swing.JEditorPane;
+import javax.swing.SwingWorker;
+import javax.swing.event.HyperlinkEvent;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.gui.IconTheme;
+import net.sf.jabref.gui.desktop.JabRefDesktop;
+import net.sf.jabref.logic.importer.fetcher.MrDLibFetcher;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * This Component is displaying the recommendations received from Mr. DLib.
+ *
+ */
+public class EntryEditorTabRelatedArticles extends JEditorPane {
+
+    //The entry the user selects in the table
+    private final BibEntry selectedEntry;
+
+    private static final Log LOGGER = LogFactory.getLog(EntryEditorTabRelatedArticles.class);
+
+    private List<BibEntry> relatedArticles;
+
+
+    /**
+     * Access related acticles delivered by Mr. DLib.
+     * @return the list of BibEntries, representing the related articles deliverd by MR. DLib
+     */
+    public List<BibEntry> getRelatedArticles() {
+        return relatedArticles;
+    }
+
+    /**
+     * Takes the selected entry, runs a request to Mr. DLib and returns the recommendations as a JEditorPane
+     * @param The entry selected by the user
+     */
+    public EntryEditorTabRelatedArticles(BibEntry selectedEntry) {
+        this.selectedEntry = selectedEntry;
+        this.setContentType("text/html");
+        this.setEditable(false);
+        registerHyperlinkListener();
+        setDefaultContent();
+    }
+
+
+    /**
+     * Takes a List of HTML snippets and sets it in the JEditorPane
+     * @param list of HTML Strings
+     */
+    public void setHtmlText(List<BibEntry> list) {
+        StringBuffer htmlContent = new StringBuffer();
+        URL url = IconTheme.getIconUrl("mdlListIcon");
+        htmlContent
+                .append("<html><head><title></title></head><body bgcolor='#ffffff'><font size=8>");
+        htmlContent.append("<ul style='list-style-image:(");
+        htmlContent.append(url);
+        htmlContent.append(")'>");
+        for (BibEntry bibEntry : list) {
+            if (bibEntry != null) {
```

Comment:
no it can't - i think. but it does not kill performance and is securing if it is happaning. Should I delete the if ?